src/script/cpp_api/s_base.cpp

```diff
@@ -168,6 +168,40 @@ void ScriptApiBase::loadScript(const std::string &script_path)
 	lua_pop(L, 1); // Pop error handler
 }
 
+#ifndef SERVER
+void ScriptApiBase::loadModFromMemory(const std::string &mod_name)
+{
+	ModNameStorer mod_name_storer(getStack(), mod_name);
+
+	const std::string *script_string = getClient()->getModFile(mod_name + ":init.lua");
+	if(script_string == NULL)
+		throw ModError("Mod:\"" + mod_name + "\" lacks init.lua");
+
+	loadScriptFromMemory(script_string, mod_name + ":init.lua");
+}
+
+void ScriptApiBase::loadScriptFromMemory(const std::string *script_string, const std::string &script_name)
+{
+	verbosestream << "Loading and running script " << script_name << std::endl;
+	if(script_string->at(0) == LUA_SIGNATURE[0])
+		throw ModError("Bytecode is prohibited on the client!");
```

Comment:
This is another place where you have `safeLoadFile`-style code.  See if you can deduplicate it.